src/reactviews/pages/ExecutionPlan/executionPlanView.ts

```diff
@@ -0,0 +1,421 @@
+		diagramNode.cost = node.cost;
+		if (node.cost) {
+			this.expensiveMetricTypes.add(ep.ExpensiveMetricType.Cost);
+		}
+
+		diagramNode.subTreeCost = node.subTreeCost;
+		if (node.subTreeCost) {
+			this.expensiveMetricTypes.add(ep.ExpensiveMetricType.SubtreeCost);
+		}
+
+		diagramNode.relativeCost = node.relativeCost;
+		diagramNode.elapsedTimeInMs = node.elapsedTimeInMs;
+		if (node.elapsedTimeInMs) {
+			this.expensiveMetricTypes.add(ep.ExpensiveMetricType.ActualElapsedTime);
+		}
+
+		let costMetrics = [];
+		for (let i = 0; node.costMetrics && i < node.costMetrics.length; ++i) {
+			costMetrics.push(node.costMetrics[i]);
+
+			this.loadMetricTypesFromCostMetrics(node.costMetrics[i].name);
+		}
+		diagramNode.costMetrics = costMetrics;
+
+		return diagramNode;
+	}
+
+	private loadMetricTypesFromCostMetrics(costMetricName: string): void {
+		if (costMetricName === 'ElapsedCpuTime') {
+			this.expensiveMetricTypes.add(ep.ExpensiveMetricType.ActualElapsedCpuTime);
+		}
+		else if (costMetricName === 'EstimateRowsAllExecs' || costMetricName === 'ActualRows') {
+			this.expensiveMetricTypes.add(ep.ExpensiveMetricType.ActualNumberOfRowsForAllExecutions);
+		}
+		else if (costMetricName === 'EstimatedRowsRead' || costMetricName === 'ActualRowsRead') {
+			this.expensiveMetricTypes.add(ep.ExpensiveMetricType.NumberOfRowsRead);
+		}
+	}
+
+	private getBadgeTypeString(badgeType: ep.BadgeType): {
+		type: string,
+		tooltip: string
+	} | undefined {
+		/**
+		 * TODO: Need to figure out if tooltip have to be removed. For now, they are empty
+		 */
+		switch (badgeType) {
+			case ep.BadgeType.Warning:
+				return {
+					type: 'warning',
+					tooltip: ''
+				};
+			case ep.BadgeType.CriticalWarning:
+				return {
+					type: 'criticalWarning',
+					tooltip: ''
+				};
+			case ep.BadgeType.Parallelism:
+				return {
+					type: 'parallelism',
+					tooltip: ''
+				};
+			default:
+				return undefined;
+		}
+	}
+
+	private populateProperties(props: ep.ExecutionPlanGraphElementProperty[] | undefined): ep.AzDataGraphCellMetric[] {
+		if (!props) {
+			return [];
+		}
+
+		props.forEach(p => {
+			this._graphElementPropertiesSet.add(p.name);
+		});
+
+		return props.filter(e => (typeof e.displayValue === 'string') && e.showInTooltip)
+			.sort((a, b) => a.displayOrder - b.displayOrder)
+			.map(e => {
+				return {
+					name: e.name,
+					value: e.displayValue,
+					isLongString: e.positionAtBottom
+				};
+			});
+	}
+
+	private populateEdges(edges: ep.InternalExecutionPlanEdge[] | undefined): ep.AzDataGraphCellEdge[] {
+		if (!edges) {
+			return [];
+		}
+
+		return edges.map(e => {
+			e.id = this.createGraphElementId();
+			return {
+				id: e.id,
+				metrics: this.populateProperties(e.properties),
+				weight: Math.max(0.5, Math.min(0.5 + 0.75 * Math.log10(e.rowCount), 6)),
+				label: ''
+			};
+		});
+	}
+
+	private createGraphElementId(): string {
+		return `element-${this.getNonce()}`;
+	}
+
+	private getNonce(): string {

```

Comment:
seems like this should be shared utility function to be done consistently across views?

Reply:
I moved it to src\reactviews\common\utils.ts